src/components/Modal/Modal.tsx

```diff
@@ -1,50 +1,69 @@
 import styles from './Modal.module.scss';
-import { PropsWithChildren } from 'react';
+import { PropsWithChildren, useState } from 'react';
 
 interface ModalProps {
   h3: string;
-  onClose: () => void;
-  onCanсel: () => void;
   onSave: () => void;
+  nameevent: string;
 }
 
 export default function Modal(props: PropsWithChildren<ModalProps>) {
+  const [modalOpen, setModalOpen] = useState(false);
+
+  const handleButtonClick = () => {
+    setModalOpen(true);
+  };
+
   return (
     <>
-      <div
-        className={styles.modalcontainer}
-        onClick={(e) => {
-          if (e.currentTarget.className === styles.modalcontainer) {
-            props.onClose();
-          }
-        }}
-      >
+      <button className={styles.buttonclock} onClick={() => handleButtonClick()}>
+        {props.nameevent}
+      </button>
+
+      {modalOpen && (
         <div
-          className={styles.modal}
+          className={styles.modalcontainer}
           onClick={(e) => {
-            e.stopPropagation();
+            if (e.currentTarget.className === styles.modalcontainer) {
```

Comment:
why is this check needed?

Reply:
The check is needed to make sure that we are clicking on the modalcontainer

Reply:
what else could we click? I thought that should be like a backdrop covering whole document, so users can't click anywhere else.

Reply:
this is a convenient UI so that the user can quickly close the modal window

Reply:
I understood and removed the unnecessary if